packages/fuel-indexer-api-server/src/uses.rs

```diff
@@ -208,132 +212,181 @@ pub(crate) async fn register_indexer_assets(
-                                    match schema_manager
-                                        .write()
-                                        .await
-                                        .new_schema(
-                                            &namespace,
-                                            &identifier,
-                                            schema,
-                                            // Only WASM can be sent over the web.
-                                            ExecutionSource::Wasm,
-                                            &mut conn,
-                                        )
-                                        .await
-                                    {
-                                        Ok(_) => {
-                                            assets.push(result);
-                                        }
-                                        Err(e) => {
-                                            let _res =
-                                                queries::revert_transaction(&mut conn)
-                                                    .await?;
-                                            return Err(e.into());
-                                        }
-                                    }
-                                }
-                                Err(e) => {
-                                    let _res =
-                                        queries::revert_transaction(&mut conn).await?;
-                                    return Err(e.into());
-                                }
-                            }
-                        }
-                    }
-                }
-                Err(e) => {
-                    let _res = queries::revert_transaction(&mut conn).await?;
-                    return Err(e.into());
+#[allow(clippy::too_many_arguments)]
+async fn register_indexer_assets_transaction(
+    conn: &mut fuel_indexer_database::IndexerConnection,
+    schema_manager: Arc<RwLock<SchemaManager>>,
+    config: IndexerConfig,
+    namespace: &str,
+    identifier: &str,
+    pubkey: &str,
+    replace_indexer: bool,
+    asset_bytes: Vec<(IndexerAssetType, Vec<u8>)>,
+) -> ApiResult<Vec<IndexerAsset>> {
+    let mut assets: Vec<IndexerAsset> = Vec::new();
+
+    let indexer_id = queries::get_indexer_id(conn, namespace, identifier).await;
+
+    // If the indexer already exists, check that the schema has not changed.
+    if let Ok(indexer_id) = indexer_id {
+        for (asset_type, data) in asset_bytes.iter() {
+            if *asset_type == IndexerAssetType::Schema {
+                // The schema must be the same. This query returns an asset
+                // if the bytes match. If it returns None (and the indexer
+                // exists), it means that its schema is different.
+                let schema = GraphQLSchema::from(data.to_vec());
+                if queries::asset_already_exists(
```

Comment:
- Not really a comment for this PR per-se
- But I (just now) confirmed that `asset_already_exists`'s `digest` resolves to the same hash as `GraphQLSchema.version()` (even though we're using we're using slightly different means of calculating the hash, which we should update/fix in https://github.com/FuelLabs/fuel-indexer/issues/1227